### PR TITLE
[FEAT] Add support function for reading request body and improve header validator

### PR DESCRIPTION
**Pull request by leske42**
So this is basically a mix of 2 PRs.
I added the support function for reading chunked and normal request body from the read buffer (this is needed for post request). For content length it reads a predefined size and for chunked it always retrieves one chunk at a time.

Screenshot from 2024-11-29 22-18-58

Also counts the chunked body size and throws a exception if payload is too large (here its limited to 9 characters for testing purposes).

Screenshot from 2024-11-29 22-20-01

I also check Expect header field which is supposed to always have "100-continue" as field value.

### CODE REVIEW

**Comment by leske42** [comment]
I forgot i had to change the Codec, because it expects to always read the whole chunked body into a str until chunk 0, before returning. This would mean storing it all in memory. Now i only read one chunk at a time, but I created a new function for this, so old version is still there, I'm just not sure we need it.

**Comment by LeaYeh** [comment]
Fuck you're genius
How you can be so productive

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,7 @@ $(NAME): cmake_build
 	@echo ${STY_GRE}${STY_BOL} "✅ Build Complete" ${STY_RES}
 	@echo ${STY_BLU} "➜ Project: " ${STY_WHI}${PROJECT}${STY_RES}
 	@echo ${STY_BLU} "➜ Status: " ${STY_GRE} "Mandatory" ${STY_RES}
+	@mv $(NAME) .
```
Comment (itislu) [inline Makefile:19]: @LeaYeh I swear you did that too, but on which branch?

Comment (leske42) [reply Makefile:19]: In your dream branch

File: include/defines.hpp [hunk 2]
```diff
@@ -147,6 +147,7 @@ enum StatusCode
     REQUEST_TIMEOUT = 408,
     NO_CONTENT_LENGTH = 411,
     PAYLOAD_TOO_LARGE = 413,
+    EXPECTATION_FAILED = 417,
```
Comment (LeaYeh) [inline include/defines.hpp:161]: We need to make decision today which one we are going to use at final

**Review by LeaYeh — APPROVED** [review]

**Review by yenthing — APPROVED** [review]

**Review by LeaYeh — APPROVED** [review]

**Review by yenthing — APPROVED** [review]

**Pull request merged**